lib/msal-common/src/utils/Constants.ts

```diff
@@ -401,3 +401,38 @@ export enum JsonTypes {
     Jwt = "JWT",
     Jwk = "JWK"
 }
+
+export enum SilentTokenRetrievalStrategy {
+    /*
+     * acquireTokenSilent will first look in the cache before going to the network with the existing refresh token.
+     * The refresh token will be renewed if it is expired.
+     * (existing default behavior)
+     */
+    CacheAndNetwork = 0, // 0 is falsy, is equivalent to not passing in a SilentTokenRetrievalStrategy
+    /*
+     * acquireTokenSilent will only look in the cache, and will not go to the network with the existing refresh token.
+     * The refresh token will not be renewed if it is expired.
+     */
+    CacheOnly = 1,
```

Comment:
Are there example scenarios for each of these options? I know that there are some scenarios where a developer might want to not use AT in the cache, and so other MSAL libraries have a `bool forceRefresh`, but I can't see when these options would be used.

Reply:
See the linked github issues for examples of applications wanting to have more control over the caching behavior.

Reply:
(those don't demonstrate all of the scenarios, but figured for completeness we would enable applications to use any of there strategies)

Reply:
I did read the associated GH issue and raised some concerns there as well - the scenario described by the user can be handled differently in my opinion.

These extensibility / configuration points increase the maintainability effort in the long run. The ROI seems poor if there aren't user scenarios or if the scenario is obscure, which I think it is.

In the end, I'm only involved, and you are [committed](https://www.visual-paradigm.com/scrum/scrum-pig-and-chicken/) ... but just thought I'd highlight that it took me a bit to find the user request and that maybe it should have been challenged more.

Reply:
Those are fair points, however, if we don't offer this, applications who truly need this will impement hacky workarounds (i.e. reading directly from the cache or manually caching tokens themselves) or not use our library at all. I'm personally an advocate for providing sensible defaults, but also allowing applications to get finer grained control if desired ("make the easy things easy and the hard things possible").

Reply:
Fair statements on both sides here.  There are a few other small differences between the WAM and JS implementations we'll have to sort out as well.  Once we have to enable the JS/WAM flow by default I expect we'll have to address or disable all of these.   @jasonnutter  do you feel strongly about fixing this?  The finer grained control get's really painful if a user had to understand differences in the JS/WAM behavior or JS w/o WAM behavior.

Reply:
Yes, I believe we need to ship it as we have partners asking for it. I think it's fine if the enums don't map perfectly for JS/WAM flow, we can figure out sensible fallbacks for scenarios that don't perfectly map.

Reply:
Catching up. IMHO, we need to advocate these as `advanced options` and have defaults defined in our docs - which could be more uniform across WAM vs non-WAM flows. Otherwise, the fine grained control may be more of a maintenance risk than help in my experience.

Reply:
Having the differences in `advanced options` doesn't help when those might behave differently depending on the client configuration which is out of the app developers control.  I've filed a consistency item with WAM so we can support this there as well.